src/ui/renderer/progress_bar.rs

```diff
@@ -0,0 +1,105 @@
+use crate::graphics::{Rectangle, Sprite, Point};
+use crate::ui::{progress_bar, Renderer};
+
+const LEFT: Rectangle<u16> = Rectangle {
+    x: 0,
+    y: 34,
+    width: 6,
+    height: 49,
+};
+
+const BACKGROUND: Rectangle<u16> = Rectangle {
+    x: LEFT.width,
+    y: LEFT.y,
+    width: 1,
+    height: LEFT.height,
+};
+
+const RIGHT: Rectangle<u16> = Rectangle {
+    x: LEFT.height - LEFT.width,
+    y: LEFT.y,
+    width: LEFT.width,
+    height: LEFT.height,
+};
+
+impl progress_bar::Renderer for Renderer {
+    fn draw(
+        &mut self,
+        bounds: Rectangle<f32>,
+        progress: f32,
+    ) {
+        let active_class = 0;
+        let background_class = 1;
+        let full = 1.0;
+        let left_width_f32 = LEFT.width as f32 / 100.0;
+        let background_width = 1.0 - 2.0 * left_width_f32;
+
+        self.sprites.add(left_sprite(bounds, background_class, full));
+        self.sprites.add(background_sprite(bounds, background_class, full));
+        self.sprites.add(right_sprite(bounds, background_class, full));
+
+        if progress > 0.0 {
+            let area = bound(progress / left_width_f32);
+            self.sprites.add(left_sprite(bounds, active_class, area));
+        }
+
+        if progress > left_width_f32 {
+            let area = bound((progress - left_width_f32) / background_width);
+            self.sprites.add(background_sprite(bounds, active_class, area));
+        }
+
+        if progress > left_width_f32 + background_width {
+            let area = bound((progress - left_width_f32 - background_width) / left_width_f32);
```

Comment:
I just noticed that calculation of `area` should be improved, cause the first and last 6% of the progress bar is relatively smaller than it should be

Reply:
it's barely noticeable so I hope it is not a blocker

Reply:
It shouldn't be an issue. We will probably change the sprites and fix the issue before release.